tests/ha/setup_hosts_and_luns.pm

```diff
@@ -53,8 +65,52 @@ sub run {
     assert_script_run "cat $mountpt/$dir_id/*.hosts >> /etc/hosts";
     assert_script_run 'cat /etc/hosts';
 
-    # Skip LUN setup in upgrades
-    return if get_var('HDDVERSION', '');
+    # Special tasks to do in upgrades
+    if (get_var('HDDVERSION', '')) {
+        # Ensure cluster is not running
+        systemctl 'stop pacemaker';
+
+        # Get IP adresses configured in /etc/corosync.conf
+        my %addr_changes = ();
+        my $old_addr     = script_output q(grep ring0 /etc/corosync/corosync.conf | awk '{print "ip:"$2}' | uniq | tr '\n' ',');
+        my $count        = 0;
+
+        foreach my $addr (split(/,/, $old_addr)) {
+            ++$count;
+            $addr =~ /ip:([\d\.]+)/;
+            $addr_changes{$1} = "%NODE$count%";
+        }
+        # Currently supporting only 2 node cluster migration
+        record_info "Cluster Nodes", "corosync.conf has $count different addresses. Only first 2 will be updated" if ($count > 2);
+
+        # Finish early if no ring0 address were found in corosync.conf.
+        if ($count == 0) {
+            systemctl 'start pacemaker';
+            return;
+        }
+
+        # Remove old addresses from HA conf files
+        replace_text_in_ha_files(%addr_changes);
+
+        # Get current IP address for both nodes. $hostname and $ipaddr already have the info for the current node
+        my $partner   = is_node(1) ? choose_node(2) : choose_node(1);
+        my $partnerip = get_ip $partner;
+
+        # Replace new IP addresses in HA conf files
+        %addr_changes = (
+            q(%NODE1%) => $ipaddr,
+            q(%NODE2%) => $partnerip
+        );
+        %addr_changes = (
+            q(%NODE1%) => $partnerip,
+            q(%NODE2%) => $ipaddr
+        ) unless (is_node(1));
+        replace_text_in_ha_files(%addr_changes);
```

Comment:
Maybe you could try to modify the configuration files only on node01 and then use `csync` to sync it on all nodes?

Reply:
Done, but had to add an additional barrier to sync when both nodes have the files fixed.

New verification runs in [node1](http://mango.suse.de/tests/1941) & [node2](http://mango.suse.de/tests/1942)